src/overlays/actors/ovl_En_Cow/z_en_cow.c

```diff
@@ -106,188 +112,215 @@ void EnCow_Init(Actor* thisx, PlayState* play) {
+        this->actionFunc = EnCow_Idle;
     }
 }
 
-void func_809DF778(EnCow* this, PlayState* play) {
+void EnCow_GiveMilkWait(EnCow* this, PlayState* play) {
     if (Actor_HasParent(&this->actor, play)) {
         this->actor.parent = NULL;
-        this->actionFunc = func_809DF730;
+        this->actionFunc = EnCow_GiveMilkEnd;
     } else {
         Actor_OfferGetItem(&this->actor, play, GI_MILK, 10000.0f, 100.0f);
     }
 }
 
-void func_809DF7D8(EnCow* this, PlayState* play) {
+void EnCow_GiveMilk(EnCow* this, PlayState* play) {
     if ((Message_GetState(&play->msgCtx) == TEXT_STATE_EVENT) && Message_ShouldAdvance(play)) {
         this->actor.flags &= ~ACTOR_FLAG_16;
         Message_CloseTextbox(play);
-        this->actionFunc = func_809DF778;
+        this->actionFunc = EnCow_GiveMilkWait;
         Actor_OfferGetItem(&this->actor, play, GI_MILK, 10000.0f, 100.0f);
     }
 }
 
-void func_809DF870(EnCow* this, PlayState* play) {
+void EnCow_CheckForEmptyBottle(EnCow* this, PlayState* play) {
     if ((Message_GetState(&play->msgCtx) == TEXT_STATE_EVENT) && Message_ShouldAdvance(play)) {
         if (Inventory_HasEmptyBottle()) {
             Message_ContinueTextbox(play, 0x2007);
-            this->actionFunc = func_809DF7D8;
+            this->actionFunc = EnCow_GiveMilk;
         } else {
             Message_ContinueTextbox(play, 0x2013);
-            this->actionFunc = func_809DF6BC;
+            this->actionFunc = EnCow_TalkEnd;
         }
     }
 }
 
-void func_809DF8FC(EnCow* this, PlayState* play) {
+void EnCow_Talk(EnCow* this, PlayState* play) {
     if (Actor_ProcessTalkRequest(&this->actor, play)) {
-        this->actionFunc = func_809DF870;
+        this->actionFunc = EnCow_CheckForEmptyBottle;
     } else {
         this->actor.flags |= ACTOR_FLAG_16;
         func_8002F2CC(&this->actor, play, 170.0f);
         this->actor.textId = 0x2006;
     }
-    func_809DF494(this, play);
+
+    EnCow_UpdateAnimation(this, play);
 }
 
-void func_809DF96C(EnCow* this, PlayState* play) {
+void EnCow_Idle(EnCow* this, PlayState* play) {
     if ((play->msgCtx.ocarinaMode == OCARINA_MODE_00) || (play->msgCtx.ocarinaMode == OCARINA_MODE_04)) {
-        if (DREG(53) != 0) {
-            if (this->unk_276 & 4) {
-                this->unk_276 &= ~0x4;
-                DREG(53) = 0;
+        // There is a complex interaction between `R_EPONAS_SONG_PLAYED` and `COW_FLAG_FAILED_TO_GIVE_MILK` to allow
+        // multiple cows to try and give milk on the same frame.
+        // `COW_FLAG_FAILED_TO_GIVE_MILK` gets set if a cow is not in range with the player to interact.
+        // In the case of a failure, `R_EPONAS_SONG_PLAYED` is not set to false incase another cow can succeed.
+        // On the following frame, if both `R_EPONAS_SONG_PLAYED` and `COW_FLAG_FAILED_TO_GIVE_MILK` are set, the
+        // first cow that updates can assume all other cows also failed and can safely unset `R_EPONAS_SONG_PLAYED`.
```

Comment:
this is a much better comment, I don't think I fully realized what was going on either (specifically ensuring `R_EPONAS_SONG_PLAYED` doesn't stay set)

```suggestion
        // multiple cows to try and give milk on the same frame.
        // `COW_FLAG_FAILED_TO_GIVE_MILK` gets set if this cow is not in range with the player to interact.
        // In the case of a failure, `R_EPONAS_SONG_PLAYED` is not set to false in case another cow can succeed.
        // On the following frame, if both `R_EPONAS_SONG_PLAYED` and `COW_FLAG_FAILED_TO_GIVE_MILK` are set, the
        // first cow that updates can assume all other cows also failed and can safely unset `R_EPONAS_SONG_PLAYED`. All cows also unset `COW_FLAG_FAILED_TO_GIVE_MILK`
```
the "this" bit as a clarification it's not checking other cows to set the flag

I suggest to add the last remark because it's not obvious `COW_FLAG_FAILED_TO_GIVE_MILK` always also ends up unset, it's in two different code paths, and if it wasn't unset properly it would clear `R_EPONAS_SONG_PLAYED` early the next time

(formatting to be fixed)

Reply:
makes sense, pushed change